python/lsst/ts/externalscripts/maintel/take_rotated_comcam.py

```diff
@@ -0,0 +1,186 @@
+                maximum: 90
+              - type: string
+          target_name:
+            description: Name of the target object.
+            type: string
+          angles:
+            description: Sequence of rotation angles to move the rotator to.
+            type: array
+            items:
+              type: number
+          slew_timeout:
+            description: Timeout for slew operations (seconds).
+            type: number
+            default: 240.0
+        required:
+          - ra
+          - dec
+          - target_name
+          - angles
+        """
+        additional_schema = yaml.safe_load(additional_schema_yaml)
+
+        schema_dict["properties"].update(additional_schema["properties"])
+
+        schema_dict.setdefault("required", [])
+        schema_dict["required"].extend(additional_schema.get("required", []))
+
+        return schema_dict
+
+    def set_metadata(self, metadata):
+        super().set_metadata(metadata)
+        metadata.coordinateSystem = MetadataCoordSys.ICRS
+        radec_icrs = ICRS(
+            Angle(self.config.ra, unit=astropy.units.hourangle),
+            Angle(self.config.dec, unit=astropy.units.deg),
+        )
+        metadata.position = [radec_icrs.ra.deg, radec_icrs.dec.deg]
+
+        # Since we are defaulting to RotType.Physical, metadata rotation system
+        # should be MOUNT
+        metadata.rotationSystem = MetadataRotSys.MOUNT
+        metadata.cameraAngle = float(self.angles[0])
+        metadata.summary = f"Rotator angles: {self.angles}"
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        await super().configure(config)
+        self.config = config
+        self.angles = config.angles
+        self.slew_timeout = getattr(config, "slew_timeout", self.slew_timeout)
+        self.log.info(f"Configured with target name={self.config.target_name}")
+
+    async def run_block(self):
+        """Execute script operations."""
+        await self.assert_feasibility()
+
+        for angle in self.angles:
+            self.log.info(
+                f"Slewing to target {self.config.target_name} with rotator angle {angle} degrees."
+            )
+
+            await self.checkpoint(
+                f"[{self.config.target_name}; "
+                f"ra={self.config.ra}, dec={self.config.dec};"
+                f"rot={angle:0.2f}]::"
+                f"Slewing to rotator angle {angle} degrees."
+            )
+
+            await self.mtcs.slew_icrs(
+                ra=self.config.ra,
+                dec=self.config.dec,
+                rot=angle,
+                rot_type=RotType.Physical,
+                target_name=self.config.target_name,
+                az_wrap_strategy=WrapStrategy.NOUNWRAP,
+            )
+
+            await self.mtcs.wait_for_inposition(
```

Comment:
you don't need to do this here.. `slew_icrs` will only finish when it is in position.